### PR TITLE
Use Typescript to lint JSDoc annotations

### PR DESCRIPTION
**Pull request by JoelEinbinder**
A re-upload of #219 from my new repository.

#65

### CODE REVIEW

File: lib/NetworkManager.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ class NetworkManager extends EventEmitter {
         response = 'ProvideCredentials';
         this._attemptedAuthentications.add(event.interceptionId);
       }
-      const {username, password} = this._credentials || {};
+      const {username, password} = this._credentials || {username: undefined, password: undefined};
```
Comment (aslushnikov) [inline lib/NetworkManager.js:119]: is this necessary now?

File: npm-debug.log [hunk 2]
```diff
@@ -0,0 +1,48 @@
+0 info it worked if it ends with ok
```
Comment (bootstraponline) [inline npm-debug.log]: I don't think we want to check in `npm-debug.log`?

**Comment by aslushnikov** [comment]
Awwwwsome.

Could you please minimize the diff (there are a lot of unnecessary changes like "!" added/removed). And it's good to go.

Also, npm-debug is not needed.

File: npm-debug.log (hunk 2 above)
Comment (plantain-00) [inline npm-debug.log]: `*.log` should be gitignored

File: lib/NetworkManager.js (hunk 1 above)
Comment (JoelEinbinder) [reply lib/NetworkManager.js:119]: Yes. I think it will help more than hurt.

**Review by aslushnikov — APPROVED** [review]

File: lib/Multimap.js [hunk 3]
```diff
@@ -14,19 +14,11 @@
...
- */
 class Multimap {
   constructor() {
-    /** @type {!Map<K, !Set<!V>>} */
     this._map = new Map();
   }
 
-  /**
-   * @param {K} key
-   * @param {V} value
```
Comment (paulirish) [inline lib/Multimap.js:28]: @JoelEinbinder just curious... TSC didn't need these annotations? And how did you know they were unnecessary ?

Comment (JoelEinbinder) [reply lib/Multimap.js:28]: It’s a bug in typescript. https://github.com/Microsoft/TypeScript/issues/18135

**Comment by stereobooster** [comment]
As soon as annotations are "compatible" with typescript it should be easy to generate TS definitions from JSDoc with something like this:
- https://github.com/brettle/jsdoc-to-typescript-declaration
- https://www.npmjs.com/package/tsd-jsdoc

**Pull request merged**